apps/src/code-studio/levels/codeStudioLevels.js

```diff
@@ -0,0 +1,33 @@
+let registeredGetResult = null;
+
+/**
+ * A number of our levels provide a function that can be used to get results.
+ * Previously this was just injected into the global namespace, making it
+ * difficult to track. This makes both the registration and the usage more explicit.
+ */
+export function registerGetResult(getResultFunction=basicGetResult) {
+  if (registeredGetResult) {
+    console.error('already have a getResult function');
+  }
+  registeredGetResult = getResultFunction;
+}
+
+export function getResult() {
+  if (!registeredGetResult) {
+    console.error('No getResult function');
+    return;
```

Comment:
Is this error case useful?

I wonder if, instead of expecting very simple level types to call `window.dashboard.codeStudioLevels.registerGetResult()` with no arguments in order to initialize things, if we should just return the `basicGetResult` result if one was never registered and omit the `registerGetResult` call entirely in those cases.

```javascript
let registeredGetResult;
export function registerGetResult(x) {
  registeredGetResult = x;
}
export function getResult() {
  if (typeof registeredGetResult === 'function') {
    return registeredGetResult();
  }
  return {
    response: 'ok',
    result: true
  };
}
```